kyaml/fix/fixsetters/fixsetters.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package fixsetters
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/setters2"
+	"sigs.k8s.io/kustomize/kyaml/setters2/settersutil"
+)
+
+// SetterFixer fixes setters in the input package
+type SetterFixer struct {
+	// Writer is the stdout to write to
+	Writer io.Writer
```

Comment:
I don't think this is the right way to do this. We should return a datastructure with the information and then leave it up to the caller to determine how it should be printed. Currently it ends up as something in between a cli-tool and a library.

Reply:
@mortent Used a buffer to return the out messages. Does this look good ?

Reply:
So changing it to a buffer is the same thing, buffer implements the `io.Writer` interface. 
I think we should try to return a struct here that contains the result of fixing the setters as proper data. And it should be returned from the `FixSetters` function. This will allow other tools than a cli to use this functionality. In the current form, it can't be easily used since the caller would have to parse the output to understand what happened.